docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,187 @@
+method calculates if a node is virtual or not and populates the SystemVendor model which is later being used by the
+assisted service to determine if a node is running on a certain provider.
+We need to make sure that the provider is listed in the 
+[isVirtual](https://github.com/openshift/assisted-installer-agent/blob/master/src/inventory/system_vendor.go#L15-L19]
+list and that it can be detected using the fields in 
+[SystemVendor](https://github.com/openshift/assisted-service/blob/master/models/system_vendor.go).
+
+#### Changes to the Openshift installer project:
+
+A small change is required in the installer project.
+Since the openshift installer is generating a random InfraID the names of the Nodes(in the manifest) are different
+from the nodes the user provisions, and since we have a provider set then the provider will try to start new machines 
+due to the naming mismatch.
+Since the InfraID is random we can't tell the user to create the Nodes with specific names.
+Also, the InfraID is later used by the platforms in various ways to identify the cluster,
+for example in oVirt we use it to set an oVirt tag which helps us group the cluster resources in oVirt,
+and we rely on it in various places such as cluster destroy or CSI/Machine provider logic to filter cluster resources
+quickly and prevent unnecessary API calls.
+
+We need to add an env var to allow us to override the generated InfraID.
+We started a PR on the [installer project](https://github.com/openshift/installer/pull/5101).
+
+### Risks and Mitigations
+
+As with every large code change, this change contains the risk of breaking existing functionality.
+This can be mitigated by sanity testing and making sure each provider implements unit tests and e2e test suite.
+
+## Design Details
+
+### Open Questions
+
+1. Throught the assisted installer code we have structs and functions that are specific to baremetal(at least from the name),
+- Are they safe to be extended to support providers?
+- Is there anything baremetal specific we should be conserned about?
+- Is the assisted installer team aims to modify the names to be generic? for example I don't see why the
+[InstallerConfig](https://github.com/openshift/assisted-service/blob/master/internal/installcfg/installcfg.go#L75-L110)
+ struct is BareMetal.
+ 
+ Some examples but it is on every file:
+- [BMACReconciler](https://github.com/openshift/assisted-service/blob/master/internal/controller/controllers/bmh_agent_controller.go#L50-L55)
+- [bareMetalInventory](https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L149-L175)
+- [InstallerConfigBaremetal](https://github.com/openshift/assisted-service/blob/master/internal/installcfg/installcfg.go#L75-L110)
+- [bmhIsMaster](https://github.com/openshift/assisted-service/blob/master/internal/ignition/ignition.go#L541-L551)
+
+2. Is there a use case in which the user will provision the Nodes on a certain provider but want to disable
```

Comment:
That is a use case that seems worth retaining. OpenShift users often do UPI installation because something about the platform integration isn't compatible with their environment or goals. AI aims to make such installation experiences much easier, and sometimes the resulting cluster will deliberately avoid enabling support for a specific platform.

Reply:
Agree, this is why the provider registration should mean that the provider is supported but the user can still choose the platform.

Reply:
I asked that above in a different conversation, one thing that is not clear to me (due to my lack of AI project knowledge probably) how will the user select the platform if there is no UI changes?

Reply:
We already have an API that returns the platforms that are suitable to the current hosts, and another API to select a platform and provide its configuration.